### ISSUE TITLE
Autostart Dark Souls

### ISSUE THREAD

**Issue by Chronial** [Wulf2k/DaS-PC-MPChan#79]
Should we add a flag to autostart Dark Souls with DSCM? That would probably make life easier for most users.

**Comment by Wulf2k** [comment]
Maybe a button.  I'm not sure on autostart.  Not opposed, just not sure.

Is it easy to do?

Keep in mind, it's not as simple as launching the EXE.  Steam launches it in some unique way, or else it just immediately exits.

**Comment by Chronial** [comment]
A button would defeat the purpose: To save on clicks. Yes, it's quite easy. The trick is hidden in the shortcuts that steam creates in your startmenu. Opening `steam://rungameid/211420` will run the game via steam.

**Comment by Wulf2k** [comment]
Keep in mind the possibility of machines that could have Steam sandboxed, or just have the steam:// association broken.  As long as it handles failure properly then I'm ok with it being automatic.

**Comment by gpotter2** [comment]
It is pretty easy actually to start dark souls without the "steam://" association:
you just got to start steam.exe with the "-applaunch 211420" param.
I added the function here: https://github.com/Wulf2k/DaS-PC-MPChan/pull/84

It might also be a good idea to add an option to disable it... But as adding checkboxes again and again would have been quite strange, i didn't do that...

**Comment by Chronial** [comment]
How is it a better idea to rely on undocumented registry values instead of using the windows protocol handling?

**Comment by gpotter2** [comment]
Huh, that's what wulf2k just said:

> Keep in mind the possibility of machines that could have Steam sandboxed, or just have the steam:// association broken.

**Comment by Wulf2k** [comment]
I was thinking more of a "fail gracefully" situation than anything else.

I'm at work right now so I can't look into the code here yet.

**Comment by Chronial** [comment]
Why do you think that there could be a situation where the registry keys that you look at are set, but the registry keys required form `steam:` are not?

**Comment by gpotter2** [comment]
Well, i don't really know 
We could use the `steam:` of course, but if the user reseted the url protocols, it won't work... Moreover, it's harder to detect if link is working or not, then just launching a file...

**Comment by Chronial** [comment]
I think this is blocked by #94.
When we add this, we should probably also add a `close with Dark Souls` option. The question is how we should handle updates in that case. Due to DSCM-net we very much care that everybody is on the latest version, and if we autostart and -close with DS, nobody will see the update notices.

**Comment by Wulf2k** [comment]
I'm still not convinced this is a necessary feature.

What are your thoughts after a week+ of pondering it?

**Comment by Chronial** [comment]
I think it would be nice to have, but rather low priority.

**Comment by Chronial** [comment]
The people are hungry for it :)
https://www.reddit.com/r/darksouls/comments/4o26oj/is_there_a_way_to_run_dscm_automatically_when_i/
https://www.reddit.com/r/darksouls/comments/4nk0ta/tutorial_how_to_launch_dscm_dark_souls_with_one/

**Comment by karasuhebi** [comment]
Yeah I'll throw in my two cents and say that I'd like to see this feature implemented. :)

**Comment by gpotter2** [comment]
This function is still useable https://github.com/Wulf2k/DaS-PC-MPChan/pull/84

**Comment by darkarrow** [comment]
If you do choose to implement this I would ask that you make it optional. Because I play Dark Souls using steam in-home streaming currently and it auto-starting the game wouldn't be beneficial at all.
